### PR TITLE
Bump WC tested up to attribute to 5.9

### PR DESCRIPTION
**Pull request by jrodger**
#### Changes proposed in this Pull Request

* GlobalStep have done a round of manual testing on WooCommerce 5.9, so let's bump the tested up to attribute (We do have [one issue](https://github.com/Automattic/woocommerce-payments/issues/3333) that might be 5.9 related, but we're aiming to have that fixed ahead of release).
* The versions of WooCommerce tested against in our CI pipeline have also been updated.

#### Testing instructions

* CI tests should pass.

### CODE REVIEW

**Review by htdat — APPROVED** [review]
LGTM! But it would be great if a second set of eyes have a look too.

File: .github/workflows/compatibility.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
       fail-fast:    false
       max-parallel: 10
       matrix:
-        woocommerce: [ '4.4.0', '4.5.0', '4.6.0', '4.7.0', '4.8.0', '4.9.0', '5.0.0', '5.1.0', '5.2.0', '5.3.0', '5.4.0', '5.5.0', '5.6.0', '5.7.0' ]
+        woocommerce: [ '4.4.0', '4.5.0', '4.6.0', '4.7.0', '4.8.0', '4.9.0', '5.0.0', '5.1.0', '5.2.0', '5.3.0', '5.4.0', '5.5.0', '5.6.0', '5.7.0', '5.8.0', '5.9.0' ]
```
Comment (vbelolapotkov) [inline .github/workflows/compatibility.yml:14]: That list looks scary, but I don't have good suggestions.

**Review by vbelolapotkov — APPROVED** [review]
LGTM! :shipit:

File: .github/workflows/compatibility.yml (hunk 1 above)
Comment (jrodger) [reply .github/workflows/compatibility.yml:14]: My suggestion is we start bringing that inline with our L-2 support policy starting in the New Year.

Comment (vbelolapotkov) [reply .github/workflows/compatibility.yml:14]: Sounds like a good plan.

**Pull request merged**